homeassistant/components/tado/tado_connector.py

```diff
@@ -68,6 +70,13 @@ def setup(self):
         tado_home = self.tado.get_me()["homes"][0]
         self.home_id = tado_home["id"]
         self.home_name = tado_home["name"]
+        self.is_x = self.tado.http.isX
+        [device.update(is_x=self.is_x) for device in self.devices]
```

Comment:
This code is strange...
Why can't the library identify if the device is a TadoX or not??? I recommend to extend the library to add support to identify if a device is a TadoX one instead of doing here this hacky workaround

Also, you are creating a list here, which is never to be used...

Reply:
The library identifies if the device is a TadoX. The `isX` attribute comes from it. 

The smell here is the leak of that `http` object. HA shouldn't be aware of the libraries internals.